syntax = "proto3";

package trap_protocol;

message TrapProtocolMsg {
  oneof inner {
    GetSessionsMsg getSessionMsg = 1;
    NewSessionMsg newSessionMsg = 2;
    DeleteSessionMsg deleteSessionMsg = 3;
    SessionDetailsMsg sessionDetailsMsg = 4;
    GetDetectionsMsg getDetectionMsg = 5;
    DetectionMsg detectionMsg = 6;
    GetStateMsg getStateMsg = 7;
    SetStateMsg setStateMsg = 8;
    StateMsg stateMsg = 9;
    GetSettingsMsg getSettingsMsg = 10;
    SetSettingsMsg setSettingsMsg = 11;
    SettingsMsg settingsMsg = 12;
  }
}

message GetSessionsMsg {}
message NewSessionMsg {
    string session = 1;
}
message DeleteSessionMsg {
    string session = 1;
}
message SessionDetailsMsg {
    string session = 1;
    int32 detections = 2;
}

message GetDetectionsMsg {
    string session = 1;
}

message DetectionMsg {
    string session = 1;
    int32 detection = 2;
    int64 created = 3;
    int64 updated = 4;
    float score = 5;
    int32 clazz = 6;
    int32 width = 7;
    int32 height = 8;
    optional bytes image = 9;
}

message GetStateMsg {}
message SetStateMsg {
    int32 active_flow = 1;
}
message StateMsg {
    int32 active_flow = 1;
}

message GetSettingsMsg {}
message SetSettingsMsg {
    SettingsMsg settings = 1;
}
message SettingsMsg {
  string trap_name = 1;
  string wifi_ssid = 2;
  string wifi_password = 3;
  bool wifi_enabled = 4;
  int32 max_sessions = 5;
  float min_score = 6;
}



